src/hotspot/cpu/x86/vm_version_x86.hpp

```diff
@@ -686,7 +686,7 @@ class VM_Version : public Abstract_VM_Version {
   static bool supports_avx256only()   { return (supports_avx2() && !supports_evex()); }
   static bool supports_avxonly()      { return ((supports_avx2() || supports_avx()) && !supports_evex()); }
   static bool supports_sha()          { return (_features & CPU_SHA) != 0; }
-  static bool supports_fma()          { return (_features & CPU_FMA) != 0 && supports_avx(); }
```

Comment:
https://github.com/openjdk/jdk/blob/53ca75b18ea419d469758475fac8352bf915b484/src/hotspot/cpu/x86/vm_version_x86.cpp#L1154-L1158
implies fma intrinsic can be used without AVX

Reply:
https://bugs.openjdk.org/browse/JDK-8181616  added support_avx() check because new Fma vectorization needs AVX: https://cr.openjdk.org/~vdeshpande/8181616/webrev.01/
Then we hit bug https://bugs.openjdk.org/browse/JDK-8182114 and bandaid it by restoring UseSSE check.
That change came before 8296168 which switch off UseAVX if UseSSE < 4:
https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/vm_version_x86.cpp#L908

This FMA check happens after UseSSE and UseAVX are set. I suggest to remove UseSSE check here instead and keep support_avx().

Reply:
Thanks for the references! I have reverted this change and will adjust the Graal intrinsic accordingly.